src/transformers/modeling_utils.py

```diff
@@ -147,6 +150,55 @@ def find_tensor_attributes(module: nn.Module) -> List[Tuple[str, Tensor]]:
         return first_tuple[1].dtype
 
 
+def get_parameter_dtype(parameter: Union[nn.Module, GenerationMixin, "ModuleUtilsMixin"]):
+    """
+    Returns the first found floating dtype in parameters if there is one, otherwise returns the last dtype it found.
+    """
+    try:
+        for t in parameter.parameters():
+            if t.is_floating_point():
+                return t.dtype
+        # if no floating dtype was found return whatever the first dtype is
+        return t.dtype
```

Comment:
btw, is this kosher python when we use what was supposed to be a local to the loop variable

I think the kosher python is to use the little-known for/else construct:

```
        for t in parameter.parameters():
            if t.is_floating_point():
                return t.dtype
        # if no floating dtype was found return whatever the first dtype is
        else:
            return t.dtype
```

no? the `t` is still a local variable to the reader.

I could be wrong. 

whatever works.

Reply:
`for/else` is like `if/else` which means that the loop fell through and didn't exit early via `break` or `return`.

Reply:
I like `for/else`, it makes it clearer that the variable is under the local scope. Will push that change :-)